.dagger/sdk_ruby.go

```diff
@@ -0,0 +1,167 @@
+	rubyRuntimeSubdir    = "sdk/ruby/runtime"
+	rubySdkSubdir        = "sdk/ruby"
+	rubyGeneratedAPIDir  = "sdk/ruby/lib/dagger"
+	rubyGeneratedAPIPath = "sdk/ruby/lib/dagger/client.gen.rb"
+
+	rubyVersion = "3.3.6"
+)
+
+type RubySDK struct {
+	Dagger *DaggerDev // +private
+}
+
+// Lint the Ruby SDK
+func (r RubySDK) Lint(ctx context.Context) (rerr error) {
+	eg, ctx := errgroup.WithContext(ctx)
+
+	eg.Go(func() (rerr error) {
+		ctx, span := Tracer().Start(ctx, "lint the ruby source")
+		defer func() {
+			if rerr != nil {
+				span.SetStatus(codes.Error, rerr.Error())
+			}
+			span.End()
+		}()
+		_, err := r.rubyBase().WithWorkdir("/").WithExec([]string{"rubocop", "-l", "-c", filepath.Join(rubySdkSubdir, "rubocop.yml"), rubyGeneratedAPIPath}).Sync(ctx)
+		return err
+	})
+
+	eg.Go(func() (rerr error) {
+		ctx, span := Tracer().Start(ctx, "check that the generated client library is up-to-date")
+		defer func() {
+			if rerr != nil {
+				span.SetStatus(codes.Error, rerr.Error())
+			}
+			span.End()
+		}()
+		before := r.Dagger.Source().
+			// the generated file is in the same repository than some other ruby files
+			// TODO: improve to not hardcode those files
+			WithoutFiles([]string{"sdk/ruby/lib/dagger/gqlclient.rb", "sdk/ruby/lib/dagger/node.rb"})
+		after, err := r.Generate(ctx)
+		if err != nil {
+			return err
+		}
+		return dag.Dirdiff().AssertEqual(ctx, before, after, []string{
+			rubyGeneratedAPIDir,
+		})
+	})
+
+	eg.Go(func() (rerr error) {
+		ctx, span := Tracer().Start(ctx, "lint the ruby runtime, which is written in Go")
+		defer func() {
+			if rerr != nil {
+				span.SetStatus(codes.Error, rerr.Error())
+			}
+			span.End()
+		}()
+		return dag.
+			Go(r.Dagger.WithModCodegen().Source()).
+			Lint(ctx, dagger.GoLintOpts{Packages: []string{rubyRuntimeSubdir}})
+	})
+
+	return eg.Wait()
+}
+
+// Test the Ruby SDK
+func (r RubySDK) Test(ctx context.Context) (rerr error) {
+	return nil
+}
+
+// Regenerate the Ruby SDK API
+func (r RubySDK) Generate(ctx context.Context) (*dagger.Directory, error) {
+	installer, err := r.Dagger.installer(ctx, "sdk")
+	if err != nil {
+		return nil, err
+	}
+	build, err := build.NewBuilder(ctx, r.Dagger.Source())
+	if err != nil {
+		return nil, err
+	}
+
+	generated := r.rubyBase().
+		With(installer).
+		WithFile("/usr/local/bin/codegen", build.CodegenBinary()).
+		WithExec([]string{"codegen", "--lang", "ruby", "-o", path.Dir(rubyGeneratedAPIPath)}).
+		WithExec([]string{"rubocop", "-a", "-c", "rubocop.yml", rubyGeneratedAPIPath}).
+		File(rubyGeneratedAPIPath)
+	return dag.Directory().WithFile(rubyGeneratedAPIPath, generated), nil
+}
+
+// Test the publishing process
+func (r RubySDK) TestPublish(ctx context.Context, tag string) error {
+	return r.Publish(ctx, tag, true, nil, "https://github.com/dagger/dagger.git", nil)
+}
+
+// Publish the Ruby SDK
+func (r RubySDK) Publish(
+	ctx context.Context,
+	tag string,
+
+	// +optional
+	dryRun bool,
+	// +optional
+	npmToken *dagger.Secret,
```

Comment:
It's not related to Ruby?

Reply:
Not yet, the `TestPublish`, `Publish` and `Bump` have not been implemented for ruby.
So it's just copy/paste from typescript code.
I'll work on those once the rest has been improved.